ngraph/frontend/onnx_import/include/onnx_import/editor/editor.hpp

```diff
@@ -70,6 +71,20 @@ namespace ngraph
             ///                     the inputs specified in its parameter.
             void set_input_shapes(const std::map<std::string, ngraph::PartialShape>& input_shapes);
 
+            /// \brief Modifies the in-memory representation of the model by setting custom input
+            ///        values for inputs specified in the provided map.
+            ///
+            /// \note This method modifies existing initializer tensor if its name matches one of
+            ///       input_name. Otherwise it adds initializer tensor into the model.
+            ///       If input tensor of matching name is present in the model, its type and shape
+            ///       are modified accordingly.
+            ///
+            /// \param input_values A collection of pairs {input_name: new_input_values} used to
+            ///                     update the ONNX model. Initializers already existing are
+            ///                     overwritten.
+            void set_input_values(
+                const std::map<std::string, std::shared_ptr<ngraph::op::Constant>>& input_values);
```

Comment:
What is the key? Is it an operation name or tensor name?

Reply:
Keys in this map are names of graph's inputs (which are imported as ngraph::Parameter). Those inputs are converted to initializers and then imported as ngraph::Constant instead.

Reply:
I thought that editor works with native ONNX model

Reply:
It does work on the ONNX model. This method takes a map, where the keys are ONNX graph input names and those specified inputs get "frozen" (replaced with initializers). When the modified model is passed to the onnx_importer, then those things that Tomasz mentioned get translated into parameters and constants.

Reply:
Ok, but at the current moment we introduced tensor names. It means that I can use tensor names in order to freeze some data. But as I understand at the current moment this method doesn't support such behavior.

Reply:
Right, but those tensor names are set by the onnx_importer when it receives an ONNX model and translates it into an ngraph::Function. Before this process we're operating on onnx::ModelProto and onnx::GraphProto objects only, so we have to  find the to be frozen input names in the model itself. For example here https://github.com/openvinotoolkit/openvino/blob/master/ngraph/test/models/onnx/add_abc.prototxt there are 3 possible inputs(A, B and C, lines: 20, 33 and 46) that can be changed to initializers. In this case you can use those names as keys for this map.